src/libraries/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentDictionary.cs

```diff
@@ -637,19 +637,22 @@ private bool TryUpdateInternal(TKey key, int? nullableHashcode, TValue newValue,
         /// </summary>
         public void Clear()
         {
-            int locksAcquired = 0;
-            try
+            if (!IsEmpty)
```

Comment:
The general idea is fine, but this isn't the best way to do it.  IsEmpty may need to take all the locks, in which case if the dictionary isn't empty, we'll be taking all the locks, then releasing them, then immediately taking them all again to do the clear.

Instead, below after the AquireAllLocks call, it should have:
```C#
if (AreAllBucketsEmpty())
{
    return;
}
```

Reply:
Ohh ok. I will put this call after AquireAllLocks call

Reply:
So you mean rather than if(isEmpty) condition, I should check with above if the condition that you posted. Correct?

Reply:
Thanks.

Reply:
Hi @stephentoub, I am not getting what required change need to be done! Function you mentioned AreAllBucketsEmpty() does not exists on that file!

Reply:
https://github.com/dotnet/runtime/blob/76904319b41a1dd0823daaaaae6e56769ed19ed3/src/libraries/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentDictionary.cs#L1424

It's currently a local function and just needs to be moved out.

Reply:
Hi @stephentoub, very thanks for the replay. I have added commit based on the suggested change.